sdk/cosmos/platform-matrix.json

```diff
@@ -33,8 +33,7 @@
       "ProfileFlag": "-Pe2e",
       "Agent": {
         "ubuntu": { "OSVmImage": "MMSUbuntu20.04", "Pool": "azsdk-pool-mms-ubuntu-2004-general" }
-      },
-      "JavaTestVersion": "1.11"
+      }
```

Comment:
@alzimmermsft , I am wondering what other azure SDKs are testing against?
@saragluna - while supporting java version 16 or 17 is good to have, but we should still aim to test our cosmos Java SDKs with at least 8 and 11. (since most of our customers are on version 8 and 11).

Reply:
The ci pipeline will run against Java 8 and Java 11, the live tests pipeline will use Java 17. I found this https://github.com/Azure/azure-sdk-for-java/blob/main/eng/pipelines/templates/stages/platform-matrix.json, but @alzimmermsft could help confirm it. 

The problem is whether we should run the live tests against JDK8, JDK 11, and JDK 17.

Reply:
Got it, thanks for the information @saragluna , let's confirm with @alzimmermsft 
Also, I am seeing some issue on spring live tests - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1927553&view=logs&j=caa02ea9-2389-5ab0-dd04-e2bf1c48642e&t=2c713f32-d936-5c8d-942a-fcedc5ff7533&l=43
will investigate it tomorrow and will hopefully unblock this PR and prepare for merge, thanks for the patience!

Reply:
Also, looks like there are more errors related to some java modules I think - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1927553&view=ms.vss-test-web.build-test-results-tab&runId=36095886&paneView=debug&resultId=100160 
@saragluna can you please do a sanity check and see if these are also related to JDK 17 ? (Since these are in live CI tests). 
also wondering, if we really need to update live CI for azure-cosmos java SDK? May be we should just limit the live CI to JDK 17 to just spring data cosmos if we need it for spring boot 3 support, thoughts?

Reply:
I think the error is related to JDK 17 since it's using the reflection. Since spring-data-cosmos depends on azure-cosmos, so for spring-data-cosmos to support JDK 17, we should ensure that azure-cosmos supports it.

Reply:
Thanks @alzimmermsft , can you please help us with the appropriate Java version testing? Please feel free to make edits / updates to this PR to allow supporting and testing Java 8, 11 and 17.